base_decorator.rb

```diff
@@ -0,0 +1,11 @@
+require './nameable'
+class Base < Nameable
+  def initialize(nameable)
```

Comment:
- Let's add the `attr_accessor :nameable` here as this is needed in the decorator